src/foam/util/test/UIDUniquenessTest.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.util.test',
+  name: 'UIDUniquenessTest',
+  extends: 'foam.nanos.test.Test',
+
+  javaImports: [
+    'foam.nanos.logger.Logger',
+    'foam.util.UIDGenerator',
+    'java.util.concurrent.ConcurrentHashMap',
+    'java.util.concurrent.atomic.AtomicBoolean'
+  ],
+
+  properties: [
+    {
+      class: 'Int',
+      name: 'size',
+      value: 100000
+    }
+  ],
+
+  methods: [
+    {
+      name: 'runTest',
+      javaCode: `
+        UIDUniquenessTest_No_duplicate_on_same_instance(x);
+        UIDUniquenessTest_Could_generate_duplicate_on_instances_with_same_machineId(x);
+        UIDUniquenessTest_No_duplicate_on_instances_with_different_machineId(x);
+      `
+    },
+    {
+      name: 'testDuplicateFound',
+      args: [ 'Boolean expected', 'String message', 'UIDGenerator... uidGenerators' ],
+      javaCode: `
+        var uids = new ConcurrentHashMap<String, String>();
+        var duplicateFound = new AtomicBoolean(false);
+        var isInterrupted = new AtomicBoolean(false);
+        var threads = new Thread[uidGenerators.length];
+        for ( var i = 0; i < uidGenerators.length; i++ ) {
+          var uidgen = uidGenerators[i];
+          threads[i] = new Thread(() -> {
+            try {
+              for ( var j = 0; j < getSize(); j++ ) {
+                if ( duplicateFound.get() ) return;
+                var uid = uidgen.getNextString();
+                if ( ! uids.containsKey(uid) ) {
+                  uids.put(uid, uid);
+                } else {
+                  duplicateFound.set(true);
+                }
+              }
+            } catch ( Exception e ) {
+              getLogger().error(e);
+              isInterrupted.set(true);
+            }
+          });
+          threads[i].start();
+        }
+
+        // Join threads
+        for ( var t : threads ) {
+          try { t.join(); } catch ( InterruptedException e ) { /* Ignored */ }
+        }
+        if ( ! isInterrupted.get() ) {
+          test( duplicateFound.get() == expected, message);
+        } else {
+          getLogger().error("An unexpected exception was thrown. Some tests might not have been executed");
+        }
```

Comment:
could check if error then test(false, "Failed to generate UID") else test the duplicateFound.